### ISSUE TITLE
Timeline, having the timeline in the middle with events above and below (configurable)

### ISSUE THREAD

**Issue by joseramalho** [almende/vis#1193]
Hi guys,

Is it possible to configure the timeline to display the timeline axis in the middle of the timeline and configure the events to display above or below he timeline, having a setting in the event?
It would be a great feature.

Cheers,
Jose

**Comment by josdejong** [comment]
This is not supported out of the box but you could do two things:

- Dig around in the source code of the Timeline and adjust it such that you can support two ItemSets and a TimeAxis in the middle.
- A simpler solution would be to create two Timelines, one without visible TimeAxis, and keep their viewports synced.

**Comment by joseramalho** [comment]
I've tried both options, but 13k lines of code doesn't help for the first option and the second option was not working due to the bindings in the scrolls and zoom to keep the two timelines in sync :/
Is this option in the scope for a future release?

**Comment by josdejong** [comment]
We don't have plans to implement this ourselves, but as this is an open source project everybody could contribute with new features.

The second option should be very easy: all you have to do is create an event listener for the `rangechange` event on each of the two Timelines, and when fired, adjust the window of the other Timeline to the `start` and `end` passed by the event using `timeline.setWindow(start, end)`.

**Comment by joseramalho** [comment]
I've tried exactly that.
First I've put the callback in one of the timelines, and it was working (with a small delay, I assume that the call back was only called after a few milliseconds.
When I put it in both timelines, I got big problems, it was entering in an infinite loop triggering the call backs in both instances.

**Comment by josdejong** [comment]
Good point. The `rangechange` event is indeed also fired when the window changed programmatically via `setWindow`. You will have to check whether the `rangechange` event was caused by a user interaction or programmatic action, you can check whether the property `byUser` is `true`, and only in that case call `setWindow` of the other Timeline. Else you indeed get an infinite loop. 

By default, the viewport of a Timeline is animated to the new start/end when calling `setWindow`. You can turn animation off via `setWindow(start, end, {animation: false})`. That will remove the small delay you see for updating the second Timeline.

**Comment by yotamberk** [comment]
@joseramalho Did you figure out how to do this?

**Comment by joseramalho** [comment]
I created two instances of vis and implemented the synchronization between the two instances...

**Comment by yotamberk** [comment]
So can we close this issue or do you still want to have a Feature Request for this?

**Comment by joseramalho** [comment]
I'm not working with the tool at this point, I would like to have it in the future, but it is your backlog :)

**Comment by yotamberk** [comment]
Changing it to Feature request =)

**Comment by jgorene** [comment]
Hello,
This thread open since a while but I also find that it would be a very interesting feature !
Unfortunately, I couldn't do that because depends to my knowledges that are yet a little too basic for now ... about ;)
